server/src/main/java/org/apache/iotdb/db/query/control/TracingManager.java

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ */
+package org.apache.iotdb.db.query.control;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import org.apache.iotdb.db.conf.IoTDBConstant;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.engine.fileSystem.SystemFileFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TracingManager {
+
+  private static final Logger logger = LoggerFactory.getLogger(TracingManager.class);
+  private BufferedWriter writer;
+
+  public TracingManager(String dirName, String logFileName){
+    File performanceDir = SystemFileFactory.INSTANCE.getFile(dirName);
+    if (!performanceDir.exists()) {
+      if (performanceDir.mkdirs()) {
+        logger.info("create performance folder {}.", performanceDir);
+      } else {
+        logger.info("create performance folder {} failed.", performanceDir);
+      }
+    }
+    File logFile = SystemFileFactory.INSTANCE.getFile(dirName + File.separator + logFileName);
+
+    FileWriter fileWriter = null;
+    try {
+      fileWriter = new FileWriter(logFile, true);
+    } catch (IOException e) {
+      logger.error("Meeting error while creating TracingManager: {}", e);
+    }
+    writer = new BufferedWriter(fileWriter);
+  }
+
+  public static TracingManager getInstance() {
+    return TracingManagerHelper.INSTANCE;
+  }
+
+  public void writeQueryInfo(long queryId, String statement, int pathsNum) throws IOException {
+    StringBuilder builder = new StringBuilder("-----------------------------\n");
+    builder.append("Query Id: ").append(queryId)
+        .append(" - Query Statement: ").append(statement)
+        .append("\nQuery Id: ").append(queryId)
+        .append(" - Start time: ")
+        .append(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(System.currentTimeMillis()))
+        .append("\nQuery Id: ").append(queryId)
+        .append(" - Number of series paths: ").append(pathsNum);
+    writer.write(builder.toString());
+    writer.newLine();
+    writer.flush();
+  }
+
+  // for align by device query
+  public void writeQueryInfo(long queryId, String statement) throws IOException {
+    StringBuilder builder = new StringBuilder("-----------------------------\n");
+    builder.append("Query Id: ").append(queryId)
+        .append(" - Query Statement: ").append(statement)
+        .append("\nQuery Id: ").append(queryId)
+        .append(" - Start time: ")
+        .append(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(System.currentTimeMillis()));
+    writer.write(builder.toString());
+    writer.newLine();
+    writer.flush();
+  }
+
+  public void writePathsNum(long queryId, int pathsNum) throws IOException {
+    writer.write(new StringBuilder("Query Id: ").append(queryId)
+        .append(" - Number of series paths: ").append(pathsNum).toString());
+    writer.newLine();
+    writer.flush();
```

Comment:
same issue

Reply:
use StringBuilder to get string first (do not need to wrap with `sync`), and wrap `writer.write` with `sync`